tools/e2etesting/steps/deployplatform.yml

```diff
@@ -1,4 +1,12 @@
 steps:
+- task: UseDotNet@2
+  displayName: 'Install .NET Core SDK for building'
+  inputs:
+    packageType: sdk
+    version: 3.1.x
+    includePreviewVersions: false
+    installationPath: $(Agent.ToolsDirectory)/dotnet
```

Comment:
how come? did they change the default images again?

Reply:
This pipeline is running on self-hosted agents, that is why we have to make sure that all our dependencies are present. And I think that it is better to define our `dotnet` dependency in the pipeline and thus make sure that we get the latest one instead of installing a specific version on the host and forgetting about it.

Reply:
Sounds fair, or you check the version and only install it if its not available yet? But I guess timing is not critical here...

Reply:
The task actually checks if desired SDK is already present:

```
Starting: Install .NET Core SDK
==============================================================================
Task         : Use .NET Core
Description  : Acquires a specific version of the .NET Core SDK from the internet or the local cache and adds it to the PATH. Use this task to change the version of .NET Core used in subsequent tasks. Additionally provides proxy support.
Version      : 2.191.0
Author       : Microsoft Corporation
Help         : https://aka.ms/AA4xgy0
==============================================================================
Tool to install: .NET Core sdk version 3.1.x.
Found version 3.1.412 in channel 3.1 for user specified version spec: 3.1.x
Version: 3.1.412 was found in cache.
Creating global tool path and pre-pending to PATH.
Finishing: Install .NET Core SDK
```